scene/main/viewport.cpp

```diff
@@ -1116,10 +1166,105 @@ Listener *Viewport::get_listener() const {
 }
 
 Camera *Viewport::get_camera() const {
+#ifdef TOOLS_ENABLED
+	if (camera_override) {
+		return camera_override;
+	}
+#endif
 
 	return camera;
 }
 
+void Viewport::enable_camera_override(bool p_enable) {
+#ifdef TOOLS_ENABLED
```

Comment:
I think you are on the right track here, but I would not use a camera node.
Simply create a camera RID for the override (VisualServer::camera_create) and override the call to
VisualServer::get_singleton()->viewport_attach_camera
I think this would make it still work and would not change any game logic that requires a current camera